gnd/src/main/java/com/google/android/gnd/ui/map/ExtentSelector.java

```diff
@@ -0,0 +1,28 @@
+package com.google.android.gnd.ui.map;
+
+import android.util.Pair;
+
+import androidx.annotation.Nullable;
+
+import com.google.android.gnd.model.feature.Feature;
+import com.google.common.collect.ImmutableSet;
+
+import io.reactivex.Observable;
+
+public interface ExtentSelector extends MapProvider.MapAdapter {
+
+  Observable<Extent> getExtentSelections();
+
+  void updateExtentSelections(ImmutableSet<Extent> extents);
+
+  void renderExtentSelectionLayer();
+
+  // By default, we assume extent selectors don't care about markers.
+  default Observable<MapMarker> getMarkerClicks() {
```

Comment:
I'm not convinced this is the right object hierarchy, or that polymorphism is needed here. In the interested of time, could you add a TODO to revisit the class structure after G4G?

Reply:
Done! Yeah, everything here is sort of ad-hoc, so I would not be shocked if only a few concepts make it out of this still standing. 

Extent and Tile both seem to be the right (or close to correct) abstractions, though I'm not sure we need a separate map adapter to support the interface--I just wanted  to keep it modular here so that it's fairly inconsequential to the rest of the codebase while the code is still volatile.